sale_payment_method_automatic_workflow/model/account_invoice.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Guewen Baconnier
+#    Copyright 2015 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    @api.returns('account.move.line')
+    def _get_payment(self):
+        self.ensure_one()
+        if self.type == "out_invoice" and self.sale_ids:
+            return self.sale_ids[0].payment_ids
+        return self.env['account.move.line'].browse()
```

Comment:
I think the call to browse() does nothing here, as `self.env[modelname]` is an empty browse record already.